src/analyzer.rs

```diff
@@ -182,6 +182,44 @@ impl ImportAssertions {
       _ => None,
     }
   }
+
+  pub fn is_known_none(&self, key: &str) -> bool {
+    match self {
+      ImportAssertions::None => true,
+      ImportAssertions::Known(map) => !map.contains_key(key),
+      _ => false,
+    }
+  }
+
+  pub fn is_empty(&self) -> bool {
+    matches!(self, ImportAssertions::None)
+  }
+
+  /// Try to get a flat dictionary, successful if there are no unknowns.
+  pub fn maybe_static(&self) -> Option<HashMap<String, String>> {
+    let mut assertions = HashMap::new();
```

Comment:
If really necessary, move within the match and specify the capacity based on the existing map:

```suggestion
    let mut assertions = HashMap::with_capacity(map.len());
```